libs/gen/EnumsSourceGenerator.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Garnet;
+
+[Generator]
+public class EnumsSourceGenerator : ISourceGenerator
```

Comment:
``ISourceGenerators`` are deprecated in favor of [incremental generators](https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.md). We should use them instead. They're slightly harder to write as you need to develop a value type model to minimize performance hit to the IDE experience (i.e. there should be no syntax nodes or compilation objects which would break the incremental caching).

See more https://andrewlock.net/creating-a-source-generator-part-9-avoiding-performance-pitfalls-in-incremental-generators/

Reply:
nice catch! done!